cachito/workers/pkg_managers/general_js.py

```diff
@@ -133,8 +169,22 @@ def download_dependencies(request_id, deps, proxy_repo_url, skip_deps=None):
             for dep_pair in list(zip(output.split("\n"), dep_batch)):
                 tarball = dep_pair[0]  # e.g. angular-animations-8.2.0.tgz
                 dir_path = dep_pair[1].rsplit("@", 1)[0]  # e.g. @angular/animations
+                version = dep_pair[1].rsplit("@", 1)[1]
+
+                # In case of external dependencies, create additional intermediate
+                # parent e.g. github/<org>/<repo> or external-<repo>
+                extra_dir = ""
+                # Set up unique path for external dependencies
+                if ":" in version and version.rsplit(":", 1)[0] in prefixes:
```

Comment:
Is `rsplit` what we want here? What if the version includes `http://example.com:8080/foo...`

Also, would using regular expressions with the `re` module help parsing this?

something like

```
match = re.match(r'(?P<prefix>[^:]+):(?P<is_external>//)?(?P<suffix>[^/]+)', version)
# do some checking for the prefix
if match:
  if match.group('is_external'):
    dir_path = f"external-{dir_path}"
  else:
    extra_dir = os.path.join(match.group('prefix'),  match.group('suffix'),

```

Reply:
Excellent catch/suggestion Athos. Will make the change. Thanks!